### ISSUE TITLE
Frontend Freezes in DEV setup

### ISSUE THREAD

**Issue by Sparkier** [OpenDevin/OpenDevin#1136]
#### Describe the bug
When I run the setup described in [Development.md](https://github.com/OpenDevin/OpenDevin/blob/main/Development.md), the browser tab loads but then becomes completely unresponsive after a short while. I suspect this is due to the changes in the prod setup not being fully tested with the dev setup (see also [this issue](https://github.com/OpenDevin/OpenDevin/issues/1132)).

Altogether this makes frontend development almost impossible.

#### Setup and configuration
**Current version**:

```bash
commit 2491a3524e9b112ef9fe4d5059161ec3f85cbc69 (HEAD -> main, origin/main, origin/HEAD)
Author: Edwards.Arno <138990495+Umpire2018@users.noreply.github.com>
Date:   Mon Apr 15 22:14:37 2024 +0800

    feat: add `question.md` template for community Q&A (#1115)

    * feat: add `question.md` template for community Q&A

    * Update .github/ISSUE_TEMPLATE/question.md

    ---------

    Co-authored-by: Robert Brennan <accounts@rbren.io>
```

* Operating System:
MacOS

**Comment by akhilvc10** [comment]
@Sparkier I'm facing a similar issue. I created a GitHub issue (#1130) for this. When I commented out this specific piece of useEffect code (see snippet below), the frontend ran successfully.
https://github.com/OpenDevin/OpenDevin/blob/f0559892ab0893d9e8d1086be69b9d74accc52bf/frontend/src/components/Files.tsx#L69

**Comment by Sparkier** [comment]
Ah nice, thank you. That's a very good pointer. Seems like the file update functionality does not work anymore.

**Comment by Sparkier** [comment]
The issue seems to be that this has been changed:
https://github.com/OpenDevin/OpenDevin/blob/c5ca74f39f9e591ac1269eef02b022e313bcbe61/opendevin/config.py#L15

Previously, this was `os.path.join(os.getcwd(), 'workspace')`. Now since we're using the root folder as the workspace here, traversing and sending over the whole file tree is super slow and freezes the app.

**Comment by foragerr** [comment]
duplicate of https://github.com/OpenDevin/OpenDevin/issues/1130 ?

**Comment by Sparkier** [comment]
Yeah kinda but this has more insight into the issue.

**Comment by rbren** [comment]
This isn't really fixed--the PR above (#1138) just changed the default setup so that it doesn't trigger the bug
